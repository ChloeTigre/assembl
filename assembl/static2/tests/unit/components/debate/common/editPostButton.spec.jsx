import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import EditPostButton from '../../../../../js/app/components/debate/common/editPostButton';

describe('EditPostButton component', () => {
  it('should render an edit post button', () => {
    const handleClickSpy = jest.fn();
    const props = {
      handleClick: handleClickSpy,
      linkClassName: 'my-custom-class'
    };
    const renderer = new ShallowRenderer();
    renderer.render(<EditPostButton {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});