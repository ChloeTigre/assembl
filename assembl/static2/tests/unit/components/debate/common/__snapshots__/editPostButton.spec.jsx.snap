// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPostButton component should render an edit post button 1`] = `
<Link
  className="my-custom-class"
  onClick={[Function]}
  onlyActiveOnIndex={false}
  style={Object {}}
>
  <span
    className="assembl-icon-edit"
  />
</Link>
`;
